src/main/java/mods/railcraft/common/plugins/forge/WorldPlugin.java

```diff
@@ -48,22 +46,34 @@ public static Block getBlock(IBlockAccess world, BlockPos pos) {
 
     @Nullable
     public static TileEntity getBlockTile(IBlockAccess world, BlockPos pos) {
-        if (pos.getY() < 0)
-            return null;
-        return world.getTileEntity(pos);
+        // see flowerpot source code
+        return world instanceof ChunkCache ? ((ChunkCache) world).getTileEntity(pos, Chunk.EnumCreateEntityType.CHECK) : world.getTileEntity(pos);
```

Comment:
Shouldn't we still leave the world bounds check? The game used to crash if you passed y == -1.

Reply:
I will try to make this into an error message on log instead.